### PR TITLE
Enable offline always

### LINKED ISSUE #4205: Enable offline mode by default
As there  are no major issues when enabling offline caching we can enable it by default for all users of the desktop application when there are stored credentials.

**Acceptance criteria**
* Offline mode is turned on by default
* Setting for turning offline mode on/off is removed
* Offline db is created when users selects "store credentials"
* Offline db is created for accounts that already have stored credentials

**Tasks**
- [ ] Remove ui from Settings > Desktop to activate/deactivate offline storage
- [ ] Delete config value from desktop settings and adapt existing code

### PR DESCRIPTION
**Pull request by johnbotris**
close #4205

### CODE REVIEW

File: src/offline/CachePostLoginAction.ts [hunk 1]
```diff
@@ -15,27 +13,24 @@ export class CachePostLoginAction implements IPostLoginAction {
...
-				loadingPromises.push(this.entityClient.loadAll(CalendarEventTypeRef, groupRoot.longEvents, CUSTOM_MIN_ID).then(() => progressMonitor?.workDone(1)))
-				loadingPromises.push(this.entityClient.loadAll(CalendarEventTypeRef, groupRoot.shortEvents, CUSTOM_MIN_ID).then(() => progressMonitor?.workDone(1)))
-			}
-			await Promise.all(loadingPromises)
-			progressMonitor?.completed()
+		// 3 work to load calendar info, 2 work to load short and long events
+		const workPerCalendar = 3 + 2
+		const totalWork = this.logins.getUserController().getCalendarMemberships().length * workPerCalendar
+		const monitorHandle = this.progressTracker.registerMonitor(totalWork)
+		const progressMonitor = this.progressTracker.getMonitor(monitorHandle)
```
Comment (charlag) [inline src/offline/CachePostLoginAction.ts:27]: should unwrap here instead of soft-calling it later

File: src/offline/CachePostLoginAction.ts [hunk 2]
```diff
@@ -15,27 +13,24 @@ export class CachePostLoginAction implements IPostLoginAction {
...
+		const calendarInfos = await this.calendarModel.loadCalendarInfos(progressMonitor!)
+		const loadingPromises = []
+		for (const {groupRoot} of calendarInfos.values()) {
+			loadingPromises.push(this.entityClient.loadAll(CalendarEventTypeRef, groupRoot.longEvents, CUSTOM_MIN_ID).then(() => progressMonitor?.workDone(1)))
+			loadingPromises.push(this.entityClient.loadAll(CalendarEventTypeRef, groupRoot.shortEvents, CUSTOM_MIN_ID).then(() => progressMonitor?.workDone(1)))
 		}
+		await Promise.all(loadingPromises)
```
Comment (charlag) [inline src/offline/CachePostLoginAction.ts]: we discussed that we should not load calendars in parallel?

**Review by charlag — CHANGES_REQUESTED** [review]
Besides comments:
1. Did you want to adjust config migration?
2. We should remove translation keys too (we have mechanism for "deprecated" keys in update-translations script)

File: src/offline/CachePostLoginAction.ts (hunk 2 above)
Comment (johnbotris) [reply src/offline/CachePostLoginAction.ts]: it's in second commit

File: src/offline/CachePostLoginAction.ts (hunk 1 above)
Comment (johnbotris) [reply src/offline/CachePostLoginAction.ts:27]: there's a second commit...

**Review by charlag — APPROVED** [review]

**Comment by charlag** [comment]
Needs test notes

**Pull request merged**